com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -640,17 +654,13 @@ public bool TrySetParent(NetworkObject parent, bool worldPositionStays = true)
                 return false;
             }
 
-            if (parent == null)
```

Comment:
Well, the issue was that we were not synchronizing the WorldPositionStays parameter.  In order to fix this on both sides (parenting and de-parenting) we need to provide the same flow since `transform.SetParent` should use the same WorldPositionStays value (particularly when it is false).

Before a user would have to either use transform.parent = null (which defaults to WorldPositionStays == true) or use transform.SetParent (which bypasses some of our netcode specific checks).

This allows a user to use NetworkObject.TrySetParent for both parenting and de-parenting.